account_asset_management/models/account_asset_line.py

```diff
@@ -0,0 +1,283 @@
+                      "an associated accounting entry."))
+            previous_id = dl.previous_id.id or False
+            self.env.cr.execute(
+                "SELECT id FROM account_asset_line "
+                "WHERE previous_id = %s" % dl.id)
+            next = self.env.cr.fetchone()
+            if next:
+                next_id = next[0]
+                self.browse(next_id).write({'previous_id': previous_id})
+        return super(AccountAssetLine, self).unlink()
+
+    @api.multi
+    def write(self, vals):
+        for dl in self:
+            if vals.keys() == ['move_id'] and not vals['move_id']:
+                # allow to remove an accounting entry via the
+                # 'Delete Move' button on the depreciation lines.
+                if not self.env.context.get('unlink_from_asset'):
+                    raise exceptions.UserError(
+                        _("You are not allowed to remove an accounting entry "
+                          "linked to an asset."
+                          "\nYou should remove such entries from the asset."))
+            elif vals.keys() == ['asset_id']:
+                continue
+            elif dl.move_id and not self.env.context.get(
+                    'allow_asset_line_update'):
+                raise exceptions.UserError(
+                    _("You cannot change a depreciation line "
+                      "with an associated accounting entry."))
+            elif vals.get('init_entry'):
+                self.env.cr.execute(
+                    "SELECT id "
+                    "FROM account_asset_line "
+                    "WHERE asset_id = %s AND move_check = TRUE "
+                    "AND type = 'depreciate' AND line_date <= %s LIMIT 1",
+                    (dl.asset_id.id, dl.line_date))
+                res = self.env.cr.fetchone()
+                if res:
+                    raise exceptions.UserError(
+                        _("You cannot set the 'Initial Balance Entry' flag "
+                          "on a depreciation line "
+                          "with prior posted entries."))
+            elif vals.get('line_date'):
+                self.env.cr.execute(
+                    "SELECT id "
+                    "FROM account_asset_line "
+                    "WHERE asset_id = %s "
+                    "AND (init_entry=TRUE OR move_check=TRUE)"
+                    "AND line_date > %s LIMIT 1",
+                    (dl.asset_id.id, vals['line_date']))
+                res = self.env.cr.fetchone()
+                if res:
+                    raise exceptions.UserError(
+                        _("You cannot set the date on a depreciation line "
+                          "prior to already posted entries."))
+        return super(AccountAssetLine, self).write(vals)
+
+    @api.multi
+    def _setup_move_data(self, depreciation_date):
+        self.ensure_one()
+        asset = self.asset_id
+        move_data = {
+            'name': asset.name,
```

Comment:
it is really weird not to have the next value from the sequence of the journal here. 

cc @fclementic2c